### PR TITLE
Features

### PR DESCRIPTION
**Pull request by Guy-Gustave**
i create a new pull request because they suggested renaming de branch and another review to make another pull request.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -11,17 +10,15 @@ def show
   def new
     @event = Event.new
```
Comment (christianotieno) [inline app/controllers/events_controller.rb:11]: This action should be responsible for listing events created by the user.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)). 

- To fulfil this, we can change this to 👇 

```suggestion
    @created_events = current_user.events
```

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -11,17 +10,15 @@ def show
...
     @event = current_user.events.build(event_params)
     @event.creator_id = session[:user_id]
     if @event.save
-      flash[:success] = "Event successfully created"
+      flash[:success] = 'Event successfully created'
       redirect_to event_path(@event)
     else
-
-      flash[:error] = "Something went wrong"
+      flash[:error] = 'Something went wrong'
```
Comment (christianotieno) [inline app/controllers/events_controller.rb:21]: ## Error creating events
- [ ] We have an issue creating events. Kindly revisit your migrations and how you are defining methods on this file to track down this bug 👇 Thank you.
Screenshot 2020-09-28 211613

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -33,22 +30,10 @@ def register
       redirect_to event_path(params[:id])
     else
 
-      flash[:error] = "Something went wrong"
+      flash[:error] = 'Something went wrong'
     end
   end
```
Comment (christianotieno) [inline app/controllers/events_controller.rb:35]: ## Revamp the UI a bit.
- [ ] While at it, kindly work the UI and make sure the link to event creation only shows when the user is signed in.
- [ ] There is no need to show all users on the page. Please fix this.
- [ ] Make sure that only valid navigation links are present at the top of the page. ie, when a user is signed in, only a sign-out link should be present of the two (sign up and sign in) and not both, as well as the create event link. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/navigation_links_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by christianotieno — CHANGES_REQUESTED** [review]
# Changes Required ♻️ 
Hello @Guy-Gustave 👋 
Good job with the changes! Keep up the good work. 👏 

KIndly, let's work on the changes below 👇 

#### Christian Otieno (TSE)

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (christianotieno) [reply app/controllers/events_controller.rb:11]: ## Correction: Wrong suggested changes

- The earlier suggested changes were meant for the `show` method and not the `new` method. Apologies. 🙏

**Comment by Guy-Gustave** [comment]
after fixing all suggested corrections about UI. and logic

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Progress @Guy-Gustave :clap: However, some requirement need to be implemented in order to get approval for this project. Check :arrow_down: 

#### Behaviour required :warning: 
* [ ] Only Logged in user can create new event ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_can_create_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- use the "before_action" callback to restrict access to certain views/actions. It's causing errors :red_circle: 
* [ ] Only used routes should be defined.
#### Tests :required :warning: 

* [ ] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [ ] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [ ] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by Guy-Gustave** [comment]
Ahmed Mahfoudh,  thanx for the quick review. but I hope that the test and those callbacks are for the next project. I was reading about it today for the next project

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Guy Nigaba** 
Great job following the improvements from the last review :muscle:  :muscle:
We still need to fix some details:

- [ ] In the list of all events, each event should have a link that redirects to show event page, in this page it should be visible all the information of the event, the attendances to the event, also it should have an option the attend the event.
- [ ] The "user show" page should show the events created by the user and also the events where the user is an attendant. The user attendances should be separated by future and previous (like you did with the events).

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Comment by Guy-Gustave** [comment]
hi,  Thanx for your suggestion, i listed all attendees of the event as mentioned in odin project, but the event, I displayed all with details. thanx

**Review by rOluochKe — APPROVED** [review]
Hello @Guy-Gustave 
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**